### ISSUE TITLE
update_rules_info doesn't work properly

### ISSUE THREAD

**Issue by txdv** [koraktor/steam-condenser#85]
```
server = GoldSrcServer.new("cs.six.lt") 
server.update_rules_info
```

results in
    ./../lib/stringio_additions.rb:43:in `cstring': undefined method`[]' for nil:NilClass (NoMethodError)

**Comment by koraktor** [comment]
Okay, just tested this against the PHP version, which works. Seems to be a problem with split packets.

**Comment by koraktor** [comment]
Ruby: Fixed split packet handling

Closed by 54039ed2b48cded38ffa4c8d0b0212e71115a9fa
